app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -475,6 +483,254 @@ class HomeActivityTest {
     }
   }
 
+  @Test
+  fun testHomeActivity_allTopicsCompleted_hidesPromotedStories() {
+    storyProgressTestHelper.markFullProgressForAllTopics(
+      profileId = createProfileId(internalProfileId),
+      timestampOlderThanOneWeek = false
+    )
+    testCoroutineDispatchers.runCurrent()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          2,
+          R.id.promoted_story_list_recycler_view
+        )
+      ).check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testHomeActivity_partialProgressForFractionsAndRatios_showsRecentlyPlayedStories() {
+    val profileId = createProfileId(internalProfileId)
+    storyProgressTestHelper.markPartialTopicProgressForFractions(
+      profileId = profileId,
+      timestampOlderThanAWeek = false
+    )
+    storyProgressTestHelper.markTwoPartialStoryProgressForRatios(
+      profileId = profileId,
+      timestampOlderThanAWeek = false
+    )
+    testCoroutineDispatchers.runCurrent()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      verifyExactTextOnHomeListItemAtPosition(
+        itemPosition = 1,
+        targetViewId = R.id.recently_played_stories_text_view,
+        stringToMatch = context.getString(R.string.recently_played_stories)
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_allTopicsCompleted_displaysAllTopicsHeader() {
+    storyProgressTestHelper.markFullProgressForAllTopics(
+      profileId = createProfileId(internalProfileId),
+      timestampOlderThanOneWeek = false
+    )
+    testCoroutineDispatchers.runCurrent()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 2)
+      verifyExactTextOnHomeListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.all_topics_text_view,
+        stringToMatch = context.getString((R.string.all_topics))
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_allTopicsCompleted_displaysAllTopicCards() {
+    storyProgressTestHelper.markFullProgressForAllTopics(
+      profileId = createProfileId(internalProfileId),
+      timestampOlderThanOneWeek = false
+    )
+    testCoroutineDispatchers.runCurrent()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        // The "All Topics" section currently should display the four test topics in two rows.
+        hasGridColumnCount(2)
```

Comment:
are we using `verifyHomeRecyclerViewHasGridColumnCount` for all occurences.

Reply:
https://github.com/oppia/oppia-android/issues/2504

Reply:
@anandwana001 Can you resolve this issue? I have filed this as a separate issue to unblock veena.

Reply:
Merge the PR once you resolve this.